Tests/FluentAssertions.Specs/Exceptions/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -23,65 +25,65 @@ public void When_getting_the_subject_it_should_remain_unchanged()
+            Func<Task> testAction = () => action.Should().ThrowAsync<ArgumentException>(
+                "because we want to test the failure {0}", "message");
 
             // Assert
-            act2.Should().Throw<XunitException>();
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
         }
 
         [Fact]
-        public void When_subject_is_null_when_expecting_an_exception_it_should_throw()
+        public async Task When_subject_is_null_when_not_expecting_a_generic_exception_it_should_throw()
         {
             // Arrange
             Func<Task> action = null;
 
             // Act
-            Func<Task> testAction = () => action.Should().ThrowAsync<ArgumentException>(
+            Func<Task> testAction = () => action.Should().NotThrowAsync<ArgumentException>(
                 "because we want to test the failure {0}", "message");
 
             // Assert
-            testAction.Should().Throw<XunitException>()
+            await testAction.Should().ThrowAsync<XunitException>()
                 .WithMessage("*because we want to test the failure message*found <null>*");
         }
 
         [Fact]
-        public void When_subject_is_null_when_not_expecting_a_generic_exception_it_should_throw()
+        public async Task When_subject_is_null_when_not_expecting_an_exception_it_should_throw()
         {
             // Arrange
             Func<Task> action = null;
 
             // Act
-            Func<Task> testAction = () => action.Should().NotThrowAsync<ArgumentException>(
+            Func<Task> testAction = () => action.Should().NotThrowAsync(
                 "because we want to test the failure {0}", "message");
 
             // Assert
-            testAction.Should().Throw<XunitException>()
+            await testAction.Should().ThrowAsync<XunitException>()
                 .WithMessage("*because we want to test the failure message*found <null>*");
         }
 
         [Fact]
-        public void When_subject_is_null_when_not_expecting_an_exception_it_should_throw()
+        public async Task When_async_method_throws_an_empty_AggregateException_it_should_fail()
         {
             // Arrange
-            Func<Task> action = null;
+            Func<Task> act = () => throw new AggregateException();
 
             // Act
-            Func<Task> testAction = () => action.Should().NotThrowAsync(
-                "because we want to test the failure {0}", "message");
+            Func<Task> act2 = () => act.Should().NotThrowAsync();
 
             // Assert
-            testAction.Should().Throw<XunitException>()
-                .WithMessage("*because we want to test the failure message*found <null>*");
+            await act2.Should().ThrowAsync<XunitException>();
         }
 
-        [Fact]
-        public async Task When_async_method_throws_an_empty_AggregateException_it_should_fail()
+        [UIFact]
```

Comment:
So why do we still need this?

Reply:
These tests a duplicated tests. There are always two tests with the **same test content** but **different thread context**.

Reply:
But why do you need this now?

Reply:
To ensure that the library is working fine with and without "valid" synchronization context.

Reply:
Can a test be annotated with both `[Fact]` and `[UIFact]` to avoid duplicating the content?
I have no strong opinions on which path we take, so just asking for now.

Reply:
No, at least not with the current version of StaFact:

```
1>...\AsyncFunctionExceptionAssertionSpecs.cs(86,27,86,107): error xUnit1013: Public method 'When_async_method_throws_an_empty_AggregateException_on_UI_thread_it_should_fail' on test class 'AsyncFunctionExceptionAssertionSpecs' should be marked as a Fact.
1>...\AsyncFunctionExceptionAssertionSpecs.cs(74,27,74,94): error xUnit1002: Test methods cannot have multiple Fact or Theory attributes
```